src/core/batchlan.cpp

```diff
@@ -1304,6 +1304,12 @@ HBLObjectRef       _ExecutionList::Execute     (_ExecutionList* parent, bool ign
 
     hy_env::EnvVariableSet(hy_env::path_to_current_bf, &cfp, true);
 
+    if (!hy_env::EnvVariableGet(hy_env::path_to_initial_bf, STRING)) {
+        hy_env::EnvVariableSet(hy_env::path_to_initial_bf, &cfp, true);
```

Comment:
Does this need to be done 2x? The same exact call occurs in line 1305 and if it's not a string after that, it won't be in 1308. Also, _ExecutionList::Execute can be called with files that have no associated paths (e.g. constructed from text), so you need to check that the string is not empty.

Reply:
1305 sets the variable for the currently executing batch file. Lines 1307-1308 sets a different variable that only executes once with the intent that it would be the top-level batch file. I'll put in some more error checking here.